### PR TITLE
Allow the PMI URL to directly contain the PMM

### LINKED ISSUE #10: reconsider extra level of indirection for retrieving a manifest?
This issue came up in the TAG during discussion of the TAG review in w3ctag/design-reviews#162.

One of the things that bothered a number of us about the specification was that the process for retrieving a manifest from a payment method identifier was to look for a `Link` response header with `rel="payment-method-manifest"` from a `GET` or `HEAD` response on the payment method identifier URL.  This seemed bad for two reasons:
* there was an extra seemingly-unneeded level of indirection
* the process just seemed out of place in the context of the Web platform

In https://github.com/w3ctag/design-reviews/issues/162#issuecomment-298143339 @triblondon enumerated four possibilities (including the spec's current behavior):

> 1. what the spec does now, where a Link header is used from the identifier URL to find the manifest (disadvantage: extra level of indirection)
> 2. require that the URL used as an identifier point directly to the payment method manifest (disadvantage: URL used as the identifier is less human-readable)
> 3. making the URL of the manifest be a /.well-known/ URL ([RFC 5785](https://tools.ietf.org/html/rfc5785)) that can be formed from the identifier URL without any network interaction (disadvantage: has to be at that URL)
> 4. using a request header (e.g., Accept) to allow the identifier URL to produce a payment method manifest (disadvantage: requires using Vary: Accept, although that may often be needed anyway)

Within the TAG, there seemed to be the most support for (and no objection to) (2).  However, we're certainly missing some of the context that the working group has, so we wouldn't insist on such a change (certainly not at this stage).

That said, we'd be interested in understanding more about the process that led the working group to make the choice it currently did, and whether perhaps there's another option that meets everybody's requirements.

### PR DESCRIPTION
**Pull request by domenic**
Closes #10. In particular, this switches from a HEAD request to a GET request, so that we can inspect the entire body if the `Link` header is not present.

@rsolomakhin to review.

### CODE REVIEW

**Comment by rsolomakhin** [comment]
It's a bit hard to follow the respec formatting. Just to double-check: the spec now issues a GET request and looks for `Link` first. If `Link` is found, GET that, otherwise return the contents of the first GET. Is my understanding correct?

**Comment by domenic** [comment]
Yeah, that's it exactly. The body of the response gets thrown away if any Link is successfully found. I also threw in the ability to terminate the GET if you find such a Link.

Note that unparseable Link values will, in this spec, fall back to trying the body.

**Comment by rsolomakhin** [comment]
> Note that unparseable Link values will, in this spec, fall back to trying the body.

Nice touch!
